src/Component.php

```diff
@@ -99,46 +119,53 @@ public function run(): void
                                     throw new UserException(sprintf(
                                         'All slices of the sliced table "%s" must have the same header ("%s" is first different).',
                                         $sourceFile->getPathname(),
-                                        $slicedFilePart->getFilename()
+                                        $slicedFilePart->getFilename(),
                                     ));
                                     // phpcs:enable
                                 }
                             }
                         }
                     }
-                    $csv = new CsvReader($detectFile, $manifest["delimiter"], $manifest["enclosure"]);
-                    if ($parameters["columns_from"] === 'auto') {
-                        $manifest["columns"] = $this->fillHeader(array_fill(0, $csv->getColumnsCount(), ""));
-                    } elseif ($parameters["columns_from"] === 'header') {
+                    $csv = new CsvReader(
+                        $detectFile,
+                        $manifest->getDelimiter(),
+                        $manifest->getEnclosure(),
+                    );
+                    if ($parameters['columns_from'] === 'auto') {
+                        $manifest->setColumns($this->fillHeader(array_fill(0, $csv->getColumnsCount(), '')));
+                    } elseif ($parameters['columns_from'] === 'header') {
                         if (empty($csv->getHeader())) {
                             throw new UserException('Header cannot be empty.');
                         }
-                        $manifest["columns"] = $this->fillHeader($csv->getHeader());
+                        $manifest->setColumns($this->fillHeader($csv->getHeader()));
                     }
                 }
-            } catch (\Keboola\Csv\Exception $e) {
+            } catch (Exception $e) {
                 throw new UserException('The CSV file is invalid: ' . $e->getMessage());
             }
 
             (new Process([
-                "mv",
+                'mv',
                 $sourceFile->getPathname(),
-                $outputPath . "/" . $sourceFile->getBasename(),
+                $outputPath . '/' . $sourceFile->getBasename(),
             ]))->mustRun();
 
-            if (!mb_check_encoding($manifest['columns'])) {
-                throw new UserException(
-                    'Column names contain invalid characters, check that the CSV uses UTF8 encoding. Column names: ' .
-                    mb_convert_encoding(implode(', ', $manifest['columns']), 'UTF-8', 'UTF-8')
-                );
+            if ($parameters['has_header'] === false) {
```

Comment:
Tady je důležitý to propsat i do toho manifestu. Je tam setHasHeader(). Output Mapping se pak podle této hodnoto rozhoduje jestli má při zápisu do storage skipnout ten první řádek nebo ne. Když nejsou vyjmenované sloupce v manifestu, bere je z headru a ten řádek skipne automaticky. Když jsou zapíše všechny. Takže je potřeba aby uživatel mohl poct nastavit hasHeader na true a ty to jen propíšeš do manifestu. Tuhle hodnotu chceme do manifestu jen propisovat pokud to uživatel nastaví v configu, ať neměníme aktuální chování. S tím datovým CSV bych nic nedělal, tj. když to nastaví na false, tak ví že ho tam nemá, nic neodebírej, jen propiš do manifestu a pokud na true, tak taky jen propsat. Pokud nenastaví nic, tak nech nenasetovaný ať se to tam nepropisuje vůbec.

Reply:
Aha, tak to jsem špatně pochopil, přepsal jsem to. Díky